hashicorp/dev-portal
# feat: use useScroll like framer-motion

> **Note**: targets #1091, largely replaces it.

This PR replaces custom-build scroll progress update logic with a `use-scroll` hook that leverages [`@motionone/dom`](https://motion.dev/dom/scroll), in a very similar way to [how `framer-motion` uses `@motionone/dom`](https://github.com/framer/motion/blob/main/packages/framer-motion/src/value/use-scroll.ts).

## 🧪 Testing

- [ ] Visit a tutorial view, such as [/waypoint/tutorials/get-started-docker/get-started-intro][/waypoint/tutorials/get-started-docker/get-started-intro]
    - Scroll progress should be visible at the bottom of the page, and should function as expected
    - Scroll progress should remain accurate when the viewport is resized
- [ ] Visit a tutorial view with tabbed content, such as [/waypoint/tutorials/get-started-docker/get-started-install][/waypoint/tutorials/get-started-docker/get-started-install]
    - Scroll progress should function as expected, updating when tabbed content on the page changes the height of the page content
- [ ] Visit another view that uses `SidebarSidecarLayout`, such as [/waypoint/docs/intro][/waypoint/docs/intro]
    - Scroll progress should not be visible on these views. It should only be visible on tutorial views.

[/waypoint/tutorials/get-started-docker/get-started-install]: https://dev-portal-git-zstutorial-scroll-progress-fram-0dd93d-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-install
[/waypoint/tutorials/get-started-docker/get-started-intro]: https://dev-portal-git-zstutorial-scroll-progress-fram-0dd93d-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-intro
[/waypoint/docs/intro]: https://dev-portal-git-zstutorial-scroll-progress-fram-0dd93d-hashicorp.vercel.app/waypoint/docs/intro
